src/backend/timeseries/create_timeseries_table.c

```diff
@@ -0,0 +1,243 @@
+	PartitionKey partitionKey;
+
+	if (!TableEmpty(relationId))
+	{
+		ereport(ERROR, (errmsg("non-empty tables can not be converted to "
+							   "timeseries table")));
+	}
+
+	if (!PartitionedTable(relationId))
+	{
+		ereport(ERROR, (errmsg("non-partitioned tables can not be converted "
+							   "to timeseries table")));
+	}
+
+	pgPartitionedTableRelation = table_open(relationId, AccessShareLock);
+	partitionKey = RelationGetPartitionKey(pgPartitionedTableRelation);
+
+	if (partitionKey->strategy != PARTITION_STRATEGY_RANGE)
+	{
+		ereport(ERROR, (errmsg("table must be partitioned by range to convert "
+							   "it to timeseries table")));
+	}
+
+	if (partitionKey->partnatts != 1)
+	{
+		ereport(ERROR, (errmsg("table must be partitioned by single column to "
+							   "convert it to timeseries table")));
+	}
+
+	if (!CheckIntervalAlignmentWithThresholds(partitionInterval,
+											  compressionThresholdInterval,
+											  retentionThresholdInterval))
+	{
+		ereport(ERROR, (errmsg("retention threshold must be greater than "
+							   "compression threshold and compression threshold "
+							   "must be greater than partition interval")));
+	}
+
+	if (!CheckIntervalAlignnmentWithPartitionKey(partitionKey, partitionInterval))
+	{
+		ereport(ERROR, (errmsg("partition interval for table partitioned on "
+							   "date must be multiple days")));
+	}
+
+	if (compressionThresholdInterval != NULL && !CheckIntervalAlignnmentWithPartitionKey(
+			partitionKey, compressionThresholdInterval))
+	{
+		ereport(ERROR, (errmsg("compression threshold interval for table "
+							   "partitioned on date must be multiple days")));
+	}
+
+	if (retentionThresholdInterval != NULL && !CheckIntervalAlignnmentWithPartitionKey(
+			partitionKey, retentionThresholdInterval))
+	{
+		ereport(ERROR, (errmsg("retention threshold interval for table "
+							   "partitioned on date must be multiple days")));
+	}
+
+	table_close(pgPartitionedTableRelation, NoLock);
+}
+
+
+/*
+ * Create the initial pre and post make partitions for the given relation id
+ * by getting the related information from citus_timeseries_tables and utilizing
+ * create_missing_partitions
+ */
+static void
+InitiateTimeseriesTablePartitions(Oid relationId)
+{
+	bool readOnly = false;
+	StringInfo initiateTimeseriesPartitionsCommand = makeStringInfo();
+	appendStringInfo(initiateTimeseriesPartitionsCommand,
+					 "SELECT create_missing_partitions(logicalrelid, now() + partitioninterval * postmakeintervalcount, now() - partitioninterval * premakeintervalcount) from citus_timeseries.citus_timeseries_tables WHERE logicalrelid = %d;",
+					 relationId);
+
+	int spiConnectionResult = SPI_connect();
+	if (spiConnectionResult != SPI_OK_CONNECT)
+	{
+		ereport(WARNING, (errmsg("could not connect to SPI manager to "
+								 "initiate timeseries table partitions")));
+		SPI_finish();
+	}
+
+	SPI_execute(initiateTimeseriesPartitionsCommand->data, readOnly, 0);
+	SPI_finish();
+}
```

Comment:
Since it would be much complex make that calculation in C, we've followed that way. As we call that function only once while creating timeseries table, it wouldn't affect the performance as well. I've also added related comment in the code.